### DISCUSSION TITLE
EG4 LIFEPOWER4 24V 200AH protocol documentation

### DISCUSSION THREAD

**Discussion by brick-co** [mr-manuel/venus-os_dbus-serialbattery#54] [General]
Hey there.  I have 6 EG4 LIFEPOWER4 200aH batteries that I have linked together with the RS cables.  When I connect to the batteries with the comm port adapter and run the "BMS tester"  software that comes with the batteries, I am getting good data.

However when I look at the raw data coming back I cannot make heads or tails from it.  I have looked over all your documentation and I cannot see that anyone has found what exactly this means

Here is the raw data
![image](https://github.com/mr-manuel/venus-os_dbus-serialbattery/assets/175278752/79449b33-b825-403b-98a5-d361e401ffae)

And here is what that translates to:
![image](https://github.com/mr-manuel/venus-os_dbus-serialbattery/assets/175278752/aa8c15d3-5b4c-4fa8-a6b7-a3313d04b088)

I know the first byte is the header and the second byte is the battery id.  and the last byte is a check digit and a ctrl code like carriage return.  

But I have no documentation that tells me what the data between is.

Any help would be most appreciated.  I am creating my own program that reads this data and displays it in graphical form.

**Comment by mr-manuel** [comment]
Hello, I got this protocol documentation from the EG4 support. [EG4-LifePower4 - MODBUS Communication Protocol (2024.07.05).pdf](https://github.com/user-attachments/files/16172338/EG4-LifePower4.-.MODBUS.Communication.Protocol.2024.07.05.pdf)

If you have more questions it would be best, if you write the support team of the manufacturer through their website.
